src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,30 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can not save in DB ", e);
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()
+        ) {
+            Country country = session.get(Country.class, id);
+            return Optional.ofNullable(country);
+
+        } catch (Exception e) {
+            throw new DataProcessingException("Can not find an object ", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can not find a сountry with id: " + id, e);
```
fix everywhere, replace generic object with specific type, add id where possible